src/entities/inputValidator/utils/validateTokens.ts

```diff
@@ -72,27 +80,32 @@ export const validateTokensRemoveLiquidityRecovery = (
     poolState: PoolState,
 ) => {
     areTokensInArray(
+        'Remove Liquidity Recovery',
         [removeLiquidityRecoveryInput.bptIn.address],
         [poolState.address],
     );
 };
 
-export const validatePoolHasBpt = (poolState: PoolState) => {
+export const validatePoolHasBpt = (action: string, poolState: PoolState) => {
     const { tokens, address } = poolState;
     const bptIndex = tokens.findIndex((t) => t.address === address);
     if (bptIndex < 0) {
-        throw new Error(
-            'INPUT_ERROR: Pool State should have BPT token included',
+        throw inputValidationError(
+            action,
+            'poolState should have BPT token included for Composable Stable pools',
```

Comment:
this is interesting with the pass through abstraction of the `action`. i like it!

Reply:
does seem a bit weird to have it married to Composable Stable pools like this in a generic `inputValidator/utils` although i do see thats the only place it is used :thinking:

Reply:
Yeah - there are a few inconsistencies that we could improve in the input validation as a whole. I'll write a list of everything I think could be improved in an issue so we can discuss.